.github/workflows/container-image.yml

```
@@ -65,8 +68,20 @@ jobs:
           file: ./Dockerfile
           platforms: linux/amd64, linux/arm64
           push: true
+          sbom: true
+          provenance: mode=max
           tags: |
             ghcr.io/${{github.repository_owner}}/kubewarden-controller:${{ env.TAG_NAME }}
+      - name: Sign container image
+        run: |
```

I think you need to run this step only if the container image is pushed to the repository. See the previous step building the container image. It has this `if` as well. 
```suggestion
      - name: Sign container image
        if: ${{ inputs.push-image }}
        run: |
```

---

Done!